### ISSUE TITLE
--allow-empty-input CLI flag not working

### ISSUE THREAD

**Issue by yipei1028** [stylelint/stylelint#4649]
> Clearly describe the bug

While stylelint running CLI  with `--allow-empty-input`, there are still error throwed...

> Which rule, if any, is the bug related to?

None.

> What code is needed to reproduce the bug?

None.

> What stylelint configuration is needed to reproduce the bug?

None.

> Which version of stylelint are you using?

13.2.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint --allow-empty-input`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

None.

> What did you expect to happen?

No error, while no input files it should exit quietly.

> What actually happened (e.g. what warnings or errors did you get)?

Error and exit code 2. It seems like there is no flag `--aei`

---

This issue is related to #2048 #3880  #4029

**Comment by jeddy3** [comment]
@yipei1028 Thanks for the report.

I can't reproduce this locally.

```shell
% npx stylelint -v      
13.2.1
% npx stylelint "*.none" --allow-empty-input
% echo $?                                   
0
% npx stylelint "*.none" --aei              
% echo $?                     
0
% npx stylelint "*.none"                    
Error: No files matching the pattern "*.none" were found.
    at /Users/jeddy3/Projects/stylelint/lib/standalone.js:213:12
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
% echo $?               
1
```

> ANS: CLI with `stylelint --allow-empty-input`

Unless you are piping std-in to stylelint, you need to specify an input glob.

From the CLI help:

```
--allow-empty-input, --aei
  When glob pattern matches no files, the process will exit without throwing an error.
```

Using `--aei` without a glob display the CLI help and exits with `2`.

Do you have a use case where you aren't specifying a glob?

**Comment by yipei1028** [comment]
Hi @jeddy3 ,
Appreciate for your quick reply.
Looks like I taking this `--allow-empty-input` the wrong way: The `empty` word makes me think I can really put **nothing** inside.

Only running the stylelint command should display the stylelint CLI help. (I found it [here](https://github.com/stylelint/stylelint/issues/2580#issuecomment-304630308)) Correct me if I misunderstood. 

And will u consider changing the name of the option? Say something like `--allow-no-match`. Rather than allow empty, what exactly happened here is more like there is no match with your input string.

Thank you in advance. :)

```
// Not working
npx stylelint --aei 

// working
npm stylelint  "*.none" --aei 
```

**Comment by jeddy3** [comment]
> `// Not working`
> `npx stylelint --aei `

What's your use case for providing no glob? I suspect all users would provide a glob unless they are piping in from stdin.

**Comment by jeddy3** [comment]
Closing due to inactivity.

**Comment by johnnybenson** [comment]
```
yarn stylelint --config ./stylelint.config.js $(git diff --name-only --oneline --relative | egrep '\\.ts$|\\.tsx$\\.js$') --allow-empty-input
```

Glob is slower than only linting modified files.

The nested git command returns empty when there are no modified files and `--aei` still exits with `2` instead of `0`.

It should exit 0 IMO.
